pkg/reconciler/v1alpha1/broker/broker_test.go

```diff
@@ -83,891 +86,1137 @@ func init() {
+		//	},
+		//	{
+		//		Name:   "Subscription.Create error when recreating",
+		//		Scheme: scheme.Scheme,
+		//		InitialState: []runtime.Object{
+		//			makeBroker(),
+		//			makeTriggerChannel(),
+		//			makeIngressChannel(),
+		//			makeDifferentSubscription(),
+		//		},
+		//		Mocks: controllertesting.Mocks{
+		//			MockCreates: []controllertesting.MockCreate{
+		//				func(_ client.Client, _ context.Context, obj runtime.Object) (controllertesting.MockHandled, error) {
+		//					if _, ok := obj.(*v1alpha1.Subscription); ok {
+		//						return controllertesting.Handled, errors.New("test error creating Subscription")
+		//					}
+		//					return controllertesting.Unhandled, nil
+		//				},
+		//			},
+		//		},
+		//		WantEvent:  []corev1.Event{events[ingressSubscriptionCreateFailed], events[brokerReconcileError]},
+		//		WantErrMsg: "test error creating Subscription",
+		//	},
+		//	{
+		//		Name:   "Broker.Get for status update fails",
+		//		Scheme: scheme.Scheme,
+		//		InitialState: []runtime.Object{
+		//			makeBroker(),
+		//			makeTriggerChannel(),
+		//			makeIngressChannel(),
+		//		},
+		//		Mocks: controllertesting.Mocks{
+		//			MockGets: []controllertesting.MockGet{
+		//				// The first Get works.
+		//				func(innerClient client.Client, ctx context.Context, key client.ObjectKey, obj runtime.Object) (controllertesting.MockHandled, error) {
+		//					if _, ok := obj.(*v1alpha1.Broker); ok {
+		//						return controllertesting.Handled, innerClient.Get(ctx, key, obj)
+		//					}
+		//					return controllertesting.Unhandled, nil
+		//				},
+		//				// The second Get fails.
+		//				func(_ client.Client, _ context.Context, _ client.ObjectKey, obj runtime.Object) (controllertesting.MockHandled, error) {
+		//					if _, ok := obj.(*v1alpha1.Broker); ok {
+		//						return controllertesting.Handled, errors.New("test error getting the Broker for status update")
+		//					}
+		//					return controllertesting.Unhandled, nil
+		//				},
+		//			},
+		//		},
+		//		WantErrMsg: "test error getting the Broker for status update",
+		//		WantEvent:  []corev1.Event{events[brokerUpdateStatusFailed]},
+		//	},
+		//	{
+		//		Name:   "Broker.Status.Update error",
+		//		Scheme: scheme.Scheme,
+		//		InitialState: []runtime.Object{
+		//			makeBroker(),
+		//			makeTriggerChannel(),
+		//			makeIngressChannel(),
+		//		},
+		//		Mocks: controllertesting.Mocks{
+		//			MockStatusUpdates: []controllertesting.MockStatusUpdate{
+		//				func(_ client.Client, _ context.Context, obj runtime.Object) (controllertesting.MockHandled, error) {
+		//					if _, ok := obj.(*v1alpha1.Broker); ok {
+		//						return controllertesting.Handled, errors.New("test error updating the Broker status")
+		//					}
+		//					return controllertesting.Unhandled, nil
+		//				},
+		//			},
+		//		},
+		//		WantErrMsg: "test error updating the Broker status",
+		//		WantEvent:  []corev1.Event{events[brokerUpdateStatusFailed]},
+		//	},
+		//	{
+		//		Name:   "Successf
```

Comment:
I'm fine with this, just curious why create a wrapper for this?

Reply:
@vaikas-google sorry, my bad, is not ready for review yet. I'm working on the UTs... I commented out the previous ones as I'm changing them to the new arch... Should be ready by later today or tomorrow.
I'll let you know so that you and @n3wscott  can make a whole pass.
Thanks!

Reply:
Oh no not your bad, I totally missed the WIP in the subject line :(